src/DevBetterWeb.Vimeo/Services/VideoServices/CreateAnimatedThumbnailsService.cs

```diff
@@ -0,0 +1,84 @@
+
+namespace DevBetterWeb.Vimeo.Services.VideoServices;
+
+public class CreateAnimatedThumbnailsService
+{
+  private readonly GetAnimatedThumbnailService _getAnimatedThumbnailService;
+  private readonly GetStatusAnimatedThumbnailService _getStatusAnimatedThumbnailService;
+  private readonly AddAnimatedThumbnailsToVideoService _addAnimatedThumbnailsToVideoService;
+  private readonly GetVideoService _getVideoService;
+  private readonly ILogger<CreateAnimatedThumbnailsService> _logger;
+
+  public CreateAnimatedThumbnailsService(
+    GetAnimatedThumbnailService getAnimatedThumbnailService, 
+    GetStatusAnimatedThumbnailService getStatusAnimatedThumbnailService, 
+    AddAnimatedThumbnailsToVideoService addAnimatedThumbnailsToVideoService, 
+    GetVideoService getVideoService, 
+    ILogger<CreateAnimatedThumbnailsService> logger)
+  {
+    _getAnimatedThumbnailService = getAnimatedThumbnailService;
+    _getStatusAnimatedThumbnailService = getStatusAnimatedThumbnailService;
+    _addAnimatedThumbnailsToVideoService = addAnimatedThumbnailsToVideoService;
+    _getVideoService = getVideoService;
+    _logger = logger;
+  }
+
+  public async Task<AnimatedThumbnailsResponse> ExecuteAsync(long videoId, CancellationToken cancellationToken = default)
+  {
+    return await CreateAnimatedThumbnails(videoId);
+  }
+
+  private async Task<AnimatedThumbnailsResponse> CreateAnimatedThumbnails(long videoId)
+  {
+    Video video = new Video();
+    while (video == null || video.Status != "available")
+    {
+      Thread.Sleep(20 * 1000);
+      var response = await _getVideoService.ExecuteAsync(videoId.ToString());
+      video = response.Data;
+    }
+
+    var startAnimation = GetRandomStart(video.Duration > 6 ? video.Duration : 0);
+    var addAnimatedThumbnailsToVideoRequest = new AddAnimatedThumbnailsToVideoRequest(videoId, startAnimation, video.Duration >= 6 ? 6 : video.Duration);
+    var addAnimatedThumbnailsToVideoResult = await _addAnimatedThumbnailsToVideoService.ExecuteAsync(addAnimatedThumbnailsToVideoRequest);
+    var pictureId = addAnimatedThumbnailsToVideoResult?.Data?.PictureId;
+    if (string.IsNullOrEmpty(pictureId))
+    {
+      _logger.LogError($"Creating Animated Thumbnails Error!");
+      return null;
+    }
+
+    var statusAnimatedThumbnails = string.Empty;
+    var getStatusAnimatedThumbnailRequest = new GetAnimatedThumbnailRequest(videoId, pictureId);
+    while (statusAnimatedThumbnails != "completed")
+    {
+      var statusResult = await _getStatusAnimatedThumbnailService.ExecuteAsync(getStatusAnimatedThumbnailRequest);
+      if (statusResult.Code == System.Net.HttpStatusCode.InternalServerError || statusResult.Code == System.Net.HttpStatusCode.Unauthorized || statusResult.Code == System.Net.HttpStatusCode.NotFound)
+      {
+        statusAnimatedThumbnails = string.Empty;
+      }
+      else
+      {
+        statusAnimatedThumbnails = statusResult.Data.Status;
+      }
+
+      Thread.Sleep(5 * 1000);
```

Comment:
If/when we write unit tests for this, all these Thread.Sleep call would be better if they used an interface, so the tests could replace them with no-ops.

Reply:
ISleepService & SleepService on vimeo package added.